ZAPD/GameConfig.cpp

```diff
@@ -0,0 +1,138 @@
+#include "GameConfig.h"
+
+#include <functional>
+#include "Utils/Directory.h"
+#include "Utils/File.h"
+#include "Utils/Path.h"
+#include "tinyxml2.h"
+
+void GameConfig::ReadTexturePool(const std::string& texturePoolXmlPath)
+{
+	tinyxml2::XMLDocument doc;
+	tinyxml2::XMLError eResult = doc.LoadFile(texturePoolXmlPath.c_str());
+
+	if (eResult != tinyxml2::XML_SUCCESS)
+	{
+		fprintf(stderr, "Warning: Unable to read texture pool XML with error code %i\n", eResult);
+		return;
+	}
+
+	tinyxml2::XMLNode* root = doc.FirstChild();
+
+	if (root == nullptr)
+		return;
+
+	for (tinyxml2::XMLElement* child = root->FirstChildElement(); child != nullptr;
+	     child = child->NextSiblingElement())
+	{
+		if (std::string(child->Name()) == "Texture")
+		{
+			std::string crcStr = std::string(child->Attribute("CRC"));
+			fs::path texPath = std::string(child->Attribute("Path"));
+			std::string texName = "";
```

Comment:
```suggestion
			std::string texName;
```

Reply:
I worry about having an uninitialised variable floating around... is that safe? Or does `std::string` just assume this value automatically?

Reply:
std::string's default constructor constructs an empty string so this is safe.